core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -120,14 +122,42 @@ public String operation() {
     return summary;
   }
 
-  @Override
-  public List<ManifestFile> manifests() {
-    if (manifests == null) {
+  private void cacheManifests() {
+    if (allManifests == null) {
       // if manifests isn't set, then the snapshotFile is set and should be read to get the list
-      this.manifests = ManifestLists.read(manifestList);
+      this.allManifests = ManifestLists.read(manifestList);
+    }
+
+    if (dataManifests == null) {
+      this.dataManifests = ImmutableList.copyOf(Iterables.filter(allManifests,
```

Comment:
The list of manifests can contain 1-2k entries, will iterating through that list twice here have an impact on performance?

Reply:
No, this would be pretty quick.